subprojects/language-native/src/testFixtures/groovy/org/gradle/language/swift/AbstractSwiftComponentIntegrationTest.groovy

```diff
@@ -115,22 +115,72 @@ abstract class AbstractSwiftComponentIntegrationTest extends AbstractNativeLangu
                 }
             }
         """
-        settingsFile << "rootProject.name = 'project'"
+        settingsFile << "rootProject.name = '${swift4Component.projectName}'"
 
         when:
         succeeds "verifyBinariesSwiftVersion"
         fails taskNameToAssembleDevelopmentBinary
 
         then:
-        failure.assertHasDescription("Execution failed for task ':$taskNameToCompileDevelopmentBinary'.")
+        failure.assertHasDescription("Execution failed for task '$developmentBinaryCompileTask'.")
         failure.assertHasCause("swiftc compiler version '${toolChain.version}' doesn't support Swift language version '${SwiftVersion.SWIFT4.version}'")
     }
 
-    abstract String getTaskNameToCompileDevelopmentBinary()
+    def "can compile Swift 3 component on Swift 3 compiler"() {
+        Assume.assumeThat(AbstractNativeLanguageComponentIntegrationTest.toolChain.version.major, Matchers.equalTo(3))
```

Comment:
It would be nice if we were consistent in declaring the requirements for the test. For some criteria, e.g. need a swift tool chain, we use `@Requires` and for others , e.g need a swift 3 tool chain, we use `Assume`.

I'd probably go for an annotation over some opaque (to the test infrastructure) calls to `Assume`.